### PR TITLE
VXFM-9844 support static route for data networks

### PR DESCRIPTION
**Pull request by danielseankim**

### CODE REVIEW

File: manifests/route.pp [hunk 1]
```diff
@@ -60,7 +71,13 @@
     group   => 'root',
     path    => "/etc/sysconfig/network-scripts/route-${interface}",
     content => template('network/route-eth.erb'),
-    before  => File["ifcfg-${interface}"],
     notify  => Service['network'],
   }
+
+  $ipaddress.each |Integer $index, String $ipaddr| {
+    exec { "Run cmd to add ${interface} static route for ${ipaddr}":
+      command => "/usr/sbin/ip route add ${ipaddr}/${cidr[$index]} via ${gateway[$index]} dev ${interface}"
```
Comment (gavin-scott) [inline manifests/route.pp]: I don't think there are any problems here but I always hate to see bash commands built with unescaped input. Can you call `Shellwords.escape` on the arguments just to make it obvious that this doesn't have any shell injection issues?

Comment (danielseankim) [reply manifests/route.pp]: @gavin-scott Can you use Shellwords.escape in puppet manifest file? I don't think I've seen a use case like that.

File: manifests/route.pp [hunk 2]
```diff
@@ -60,7 +71,17 @@
     group   => 'root',
     path    => "/etc/sysconfig/network-scripts/route-${interface}",
     content => template('network/route-eth.erb'),
-    before  => File["ifcfg-${interface}"],
     notify  => Service['network'],
   }
+
+  $ipaddress.each |Integer $index, String $ipaddr| {
+    $ipaddr_escaped = shell_escape($ipaddr)
+    $cidr_escaped = shell_escape($cidr[$index])
```
Comment (gavin-scott) [inline manifests/route.pp]: You could combine above to like:

    $route_escaped = shell_escape("$ipaddr/$cidr[$index]")

File: manifests/route.pp [hunk 3]
```diff
@@ -60,7 +71,17 @@
...
+    $gateway_escaped = shell_escape($gateway[$index])
+
+    exec { "Run cmd to add ${interface} static route for ${ipaddr}":
+      command => "/usr/sbin/ip route add ${ipaddr_escaped}/${cidr_escaped} via ${gateway_escaped} dev ${interface}"
```
Comment (gavin-scott) [inline manifests/route.pp]: Can you escape the interface also? Makes it easier when reviewing to see everything is escaped.

File: manifests/route.pp (hunk 1 above)
Comment (danielseankim) [reply manifests/route.pp]: @gavin-scott Using built-in function `shell_escape` seems to do the job. Let me know if this looks good to you.

File: manifests/route.pp (hunk 3 above)
Comment (danielseankim) [reply manifests/route.pp]: done

File: manifests/route.pp (hunk 2 above)
Comment (danielseankim) [reply manifests/route.pp]: @gavin-scott I don't think that's possible due to puppet manifest being picky with `[ ]`.

Comment (danielseankim) [reply manifests/route.pp]: maybe I could do:
```
$cidr_at_index = $cidr[$index]
$route_escaped = shell_escape("$ipaddr/$cidr_at_index")
```

Comment (gavin-scott) [reply manifests/route.pp]: Maybe with `"${cidr[$index]}"?

Comment (danielseankim) [reply manifests/route.pp]: That did work. We'll use that.

File: manifests/route.pp [hunk 4]
```diff
@@ -42,25 +44,47 @@
...
+  $interface_escaped = shell_escape($interface)
 
-  file { "route-${interface}":
+  file { "route-${interface_escaped}":
     ensure  => 'present',
     mode    => '0644',
     owner   => 'root',
     group   => 'root',
-    path    => "/etc/sysconfig/network-scripts/route-${interface}",
+    path    => "/etc/sysconfig/network-scripts/route-${interface_escaped}",
```
Comment (gavin-scott) [inline manifests/route.pp]: Path should use `$interface`, not `$interface_escaped`. The escaping is for the shell which is not in play here.

**Pull request merged**